sharding/nodesCoordinator/indexHashedNodesCoordinatorRegistry_test.go

```diff
@@ -252,23 +252,92 @@ func TestIndexHashedNodesCoordinator_GetNodesCoordinatorRegistry(t *testing.T) {
 		}
 
 		storer := &storage.StorerStub{
-			GetCalled: func(key []byte) (b []byte, err error) {
+			SearchFirstCalled: func(key []byte) ([]byte, error) {
 				switch {
-				case bytes.Equal(append([]byte(common.NodesCoordinatorRegistryKeyPrefix), []byte(fmt.Sprint(1))...), key):
+				case bytes.Equal(append([]byte(common.NodesCoordinatorRegistryKeyPrefix), []byte(fmt.Sprint(10))...), key):
 					return nil, errors.New("first get error")
 				default:
-					return nil, errors.New("invalid key")
+					nodesConfigRegistryBytes, _ := json.Marshal(nodesConfigRegistry)
+					return nodesConfigRegistryBytes, nil
 				}
 			},
+		}
+
+		nodesConfig, err := GetNodesCoordinatorRegistry([]byte("key"), storer, 10, numStoredEpochs)
+		require.Nil(t, err)
+		require.Equal(t, nodesConfigRegistry, nodesConfig)
+	})
+
+	t.Run("getting from old key, should also save configuration with new keys", func(t *testing.T) {
+		t.Parallel()
+
+		nodesConfigRegistry := &NodesCoordinatorRegistry{
+			EpochsConfig: map[string]*EpochValidators{
+				"10": {
+					EligibleValidators: map[string][]*SerializableValidator{
+						"val1": {
+							{
+								PubKey: []byte("pubKey1"),
+							},
+						},
+					},
+				},
+				"9": {
+					EligibleValidators: map[string][]*SerializableValidator{
+						"val2": {
+							{
+								PubKey: []byte("pubKey2"),
+							},
+						},
+					},
+				},
+				"8": {
+					EligibleValidators: map[string][]*SerializableValidator{
+						"val3": {
+							{
+								PubKey: []byte("pubKey3"),
+							},
+						},
+					},
+				},
+			},
+			CurrentEpoch: 10,
+		}
+
+		numPutCalls := uint32(0)
+		storer := &storage.StorerStub{
 			SearchFirstCalled: func(key []byte) ([]byte, error) {
-				nodesConfigRegistryBytes, _ := json.Marshal(nodesConfigRegistry)
-				return nodesConfigRegistryBytes, nil
+				switch {
+				case bytes.Equal(append([]byte(common.NodesCoordinatorRegistryKeyPrefix), []byte(fmt.Sprint(10))...), key):
+					return nil, errors.New("first get error")
+				default:
+					nodesConfigRegistryBytes, _ := json.Marshal(nodesConfigRegistry)
+					return nodesConfigRegistryBytes, nil
+				}
+			},
+			PutCalled: func(key, data []byte) error {
+				switch {
+				case bytes.Equal(append([]byte(common.NodesCoordinatorRegistryKeyPrefix), []byte(fmt.Sprint(10))...), key):
```

Comment:
can we create some variables for the cases here? it is hard to read

Reply:
right, added variables